hs_core/management/commands/ingest_bag.py

```diff
@@ -0,0 +1,103 @@
+"""
+This command ingests a bag into a resource.  You can provide a resource_id to ingest the bag or the command will create
+a new resource to ingest the bag into.
+"""
+import os
+
+from django.core.exceptions import ValidationError
+from django.core.files.uploadedfile import UploadedFile
+from django.core.management.base import BaseCommand
+
+from hs_core.hydroshare import delete_resource_file, User, add_file_to_resource, create_resource
+from hs_composite_resource.models import CompositeResource
+from zipfile import ZipFile
+from pathlib import Path
+
+from hs_core.views.utils import ingest_bag
+
+
+def extract_resource_id(bag_file_path):
+    """Extracts the resource id by reading the root directory name inside the zip"""
+    with ZipFile(bag_file_path, 'r') as zip_file:
+        resource_id = Path(zip_file.namelist()[0]).parts[0]
+    return resource_id
+
+
+def is_hydrohsare_bagit(bag_file_path):
+    """Checks for a hydroshare bagit file structure in a zip"""
+    resource_id = extract_resource_id(bag_file_path)
+    bagit_files = [
+        os.path.join(resource_id, "bagit.txt"),
+        os.path.join(resource_id, "manifest-md5.txt"),
+        os.path.join(resource_id, "readme.txt"),
+        os.path.join(resource_id, "tagmanifest-md5.txt"),
+        os.path.join(resource_id, "data", "resourcemetadata.xml"),
+        os.path.join(resource_id, "data", "resourcemap.xml")
+    ]
+    with ZipFile(bag_file_path, 'r') as zip_file:
+        for bagit_file in bagit_files:
+            if bagit_file not in zip_file.namelist():
+                return False
+    return True
+
+
+class Command(BaseCommand):
+    help = "Ingest a zipped bag archive into a resource"
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's: none does nothing.
+        parser.add_argument('bag_path', type=str)
+
+        parser.add_argument(
+            '-r',
+            '--resource_id',
+            type=str,
+            help="Existing resource to ingest into. If omitted, the bag's resource_id will be used")
+
+        parser.add_argument(
+            '-o',
+            '--overwrite',
+            action='store_true',
+            help="Overwrites resource if it exists on HydroShare"
+        )
+
+        parser.add_argument(
+            '-u',
+            '--user_id',
+            type=str,
```

Comment:
user_id is integer, so ```int``` type will be more appropriate and consistent with default value type.

Reply:
you're right, fixed.